tools/src/test/python/test_command.py

```diff
@@ -199,13 +202,22 @@ def test_long_output():
         file.flush()
         assert os.path.getsize(file.name) == num_bytes
 
-        cmd = Command(["/bin/cat", file.name])
+        if shorten:
+            num_lines //= 2
+            line_length //= 2
+        cmd = Command(["/bin/cat", file.name],
+                      max_lines=num_lines, max_line_length=line_length)
         cmd.execute()
 
         assert cmd.getstate() == Command.FINISHED
         assert cmd.getretcode() == 0
         assert cmd.geterroutput() is None
-        assert len("".join(cmd.getoutput())) == num_bytes
+        assert len(cmd.getoutput()) == num_lines
+        if shorten:
+            # Add 3 for the '...' suffix.
+            assert len([len(x) for x in cmd.getoutput() if len(x) > line_length + 3]) == 0
```

Comment:
Isn't the inner `len` unnecessary here?

Something like this might be slightly more descriptive:
```python
assert all(len(x) <= line_length + 3 for x in cmd.getoutput())
```

Reply:
much nicer, thanks !